svgdigitizer/electrochemistry/electrolyte.py

```diff
@@ -0,0 +1,196 @@
+                for e in temp:
+                    q = i['concentration']['value'] * normalize_unit(i['concentration']['unit'])
+                    e.conc = q.to("mol / l").value
+                acids_bases.extend(temp)
+
+        system = System(*acids_bases)
+        system.pHsolve()
+
+        return system.pH
+
+    @classmethod
+    def lookup_CID_from_InChI(cls, InChIKey=None, InChI=None):
+        r"""
+        Lookup Pubchem CID from InChI or InChiKey.
+
+        EXAMPLES::
+
+            CID can be retrieved with InChIKey::
+            >>> from svgdigitizer.electrochemistry.electrolyte import Electrolyte
+            >>> Electrolyte.lookup_CID_from_InChI(InChIKey='NBIIXXVUZAFLBC-UHFFFAOYSA-N') # phosphoric acid
+            1004
+
+            CID can be retrieved with InChI::
+            >>> Electrolyte.lookup_CID_from_InChI(InChI='InChI=1S/H3O4P/c1-5(2,3)4/h(H3,1,2,3,4)') # phosphoric acid
+            1004
+
+        """
+        if InChIKey:
+            return pcp.get_cids(InChIKey, 'inchikey')[0]
+        elif InChI:
+            return pcp.get_cids(InChI, 'inchi')[0]
+
+    @classmethod
+    def lookup_chemical_ids_from_name(cls, chemical_name):
+        r"""
+        Lookup CID, InChI, and InChIKey from chemical name.
+
+        EXAMPLES::
+
+            CID can be retrieved with InChIKey::
+            >>> from svgdigitizer.electrochemistry.electrolyte import Electrolyte
+            >>> Electrolyte.lookup_chemical_ids_from_name('phosphoric acid')
+            {'CID': 1004, 'InChI': 'InChI=1S/H3O4P/c1-5(2,3)4/h(H3,1,2,3,4)', 'InChIKey': 'NBIIXXVUZAFLBC-UHFFFAOYSA-N'}
+
+        """
+        results = pcp.get_cids(chemical_name, 'name')
+        len_results = len(results)
+        if len_results > 1:
+            raise RuntimeError("Ambiguous chemical name! More than one chemical id return by pubchem. Use another chemical name")
+        elif len_results == 0:
+            raise RuntimeError(f"Name of chemical {chemical_name} not found in pubchem! Check spelling.")
+        else:
+            return pcp.get_properties(("InChIKey", "InChI"), results[0])[0]
+
+    def get_pKa(self, InChIKey):
+        r"""
+        Get pKa
+        Pubchem lookup only works for neutral acids!
+
+        """
+        pattern = re.compile(r'.*(?P<pKa>\d+\.\d*).*')
+        try:
+            return acid_base_dict[InChIKey]['components']
+        except KeyError:
+            cid = self.lookup_CID_from_InChI(InChIKey=InChIKey)
+            pKa_result = self.lookup_pKa(cid)['pKa']
+            # TODO support salts etc.
+            return [Acid(pKa=[pattern.match(i).group('pKa') for i in pKa_result.split(';')], charge=0)]
+
+    def lookup_pKa(self, cid):
+        r"""
+        get url from Pubchem to get pka lookup result
+        'XML' can be replaced with 'JSON' but it is harder to parse later on
+        for more info about Pubchem output types: https://pubchemdocs.ncbi.nlm.nih.gov/pug-rest$_Toc494865558
```

Comment:
```suggestion
        Return the acid dissociation constant of `cid`.
```
The comment here should probably be moved into the implementation. It is not describing the interface really but the implementation.